# Mariano Villanueva Aguilar

# 12/092024

# T04 Programación Lineal

Una multinacional farmacéutica desea fabricar un compuesto nutritivo a base de dos productos A y B.  El producto A contiene  $30\%$ de proteínas,  un $1\%$ de grasas y un  $10\%$ de azúcares.  El producto B  contiene  un  $5\%$  de proteínas,  un  $7\%$ de grasas  y un  $10\%$  de  azúcares.

El compuesto debe de tener,  al menos, $25g$ de proteínas,  $6g$ de grasas y $30g$ de azúcares.  El coste del producto A es de $0.6$ u.m./g  y el de B es de $0.2$  u.m./g

Se desea encontrar la cantidad en gramos de cada producto para que el coste total sea mínimo.

Formular el problema de decidir cuánto hacer de cada producto como un problema de programación lineal.

1.Criterio de optimización

Minimizar costo

2. ¿Sobre que puedo influir? y variables

En la cantidad de producto A y cantidad de producto B
$$ x_A,x_B $$
3. Función de costo
$$ Costo=0.6x_A + 0.2x_B $$
4. Información adicional
$$ proteina \geq 25$$
$$ grasas \geq 6$$
$$ azucar \geq 30$$

Proteína
$$ 0.3x_A+0.05x_B \geq 25 $$
Grasas
$$ 0.01x_A+0.07x_B \geq 6 $$
Azúcar
$$ 0.1x_A+0.1x_B \geq 30 $$

5. Vectorizar en forma

In [57]:
import numpy as np
import scipy.optimize as opt
c=np.array([0.6,0.2])
Aub=np.array([
    [-0.3,-0.05],
    [-0.01,-0.07],
    [-0.1,-0.1]
])
bub=np.array([
    [-25],
    [-6],
    [-30],
])



In [7]:
sol=opt.linprog(c,Aub,bub, integrality=1)
sol

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: 76.00000000000001
              x: [ 4.000e+01  2.600e+02]
            nit: -1
          lower:  residual: [ 4.000e+01  2.600e+02]
                 marginals: [ 0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf]
                 marginals: [ 0.000e+00  0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 3.553e-15  1.260e+01  7.105e-15]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00]
 mip_node_count: 1
 mip_dual_bound: 76.00000000000001
        mip_gap: 0.0

In [9]:
sol.x

array([ 40., 260.])

In [11]:
Aub@sol.x

array([-25. , -18.6, -30. ])

Se toman 40 g del producto A y 260g del producto B

Una  fábrica  de  carrocerías  de  automóviles  y camiones  tiene  dos naves. 

+ En  la nave A,  para  hacer  la  carrocería  de  un  camión,  se  invierten  siete  días-operario,  para  fabricar  la  de  un  coche  se  precisan dos días-operario.

+ En la nave B se invierten tres días operario tanto en carrocerías de camión como de coche. 

Por limitaciones de mano de obra y maquinaria, la nave A dispone de $300$ días operario, y la nave B de $270$ días-operario.

Si los beneficios que se obtienen por cada camión son de $600$ u.m y por  cada  automóvil  $200$ u.m,  ¿cuántas  unidades  de  cada  uno  se deben  producir  para maximizar las  ganancias? 

**Nota:** el problema $\max_{\boldsymbol{x}}\boldsymbol{g}(\boldsymbol{x})$ es equivalente a $\min_{\boldsymbol{x}}-\boldsymbol{g}(\boldsymbol{x})$.

1. Criterio de optimización
Maximizar ganancia

2. ¿En qué puedo infuir?
Cantidad de autos en A $$ a_A $$
Cantidad de camiones en A $$ c_A $$
Cantidad de autos en B $$ a_B $$
Cantidad de camiones en B $$ c_B $$

3. Función de Costo
$$ Costo = 200a_A+600c_A+200a_B+600c_B $$
4. Info adicional
Nave A
$$ 2a_A+7cA \leq 300 $$
Nave B
$$ 3a_B +3c_B \leq 270 $$

5. Vectorizar en forma

In [35]:
import numpy as np
import scipy.optimize as opt
c2=np.array([-200,-600,-200,-600])
Aub2=np.array([
    [2,7,0,0],
    [0,0,3,3],
])
bub2=np.array([
    [300],
    [270],
])


In [37]:
sol2=opt.linprog(c2,Aub2,bub2, integrality=1)
sol2

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -84000.0
              x: [ 1.500e+02 -0.000e+00  0.000e+00  9.000e+01]
            nit: -1
          lower:  residual: [ 1.500e+02 -0.000e+00  0.000e+00  9.000e+01]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf        inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 0.000e+00  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00]
 mip_node_count: 1
 mip_dual_bound: -84000.0
        mip_gap: 0.0

In [39]:
sol2.x

array([150.,  -0.,   0.,  90.])

Se elaboran 150 carros en la planta A y 90 camiones en la planta B

Se gana 84,000

Una  empresa tiene dos fábricas A y B.  En ellas  se fabrica un determinado  producto, a razón de 500  y 400 unidades por día respectivamente.  El producto  ha  de  ser distribuido posteriormente  a  tres  centros  C,  D  y  E,  que  requieren,  respectivamente,  200,  300  y  400  unidades.  Los  costos  de  transportar cada unidad del producto desde cada fábrica a cada centro distribuidor son los  indicados  en la tabla  siguiente: 

Factoría|C|D|E|Fabricación (Unidades)
:----|----|----|----|----
A|  50 u.m|60 u.m|10 u.m|500 u
B|  25 u.m|40 u.m|20 u.m|400 u
Demanda|200|300|400|

**¿De qué manera  deben  organizar  el transporte a fin de que los gastos sean mínimos?**


Resolver con `linprog`

1. Criterio de optimización
Minimizar los gastos por transporte

2. ¿En qué puedo infuir?
Cantidad de productos A y cantidad de productos de B a través de los medios de transporte
$$ A_C,A_D,A_E,B_C,B_D,B_E$$


3. Función de Costo
$$ Costo = 50A_C+25B_C+60A_D+40B_D+10A_E+20B_E $$

4. Info adicional

Cantidad de producto que manda A
$$A_C+A_D+A_E = 500 $$
Cantidad de producto que manda B
$$ B_C+B_D+B_E = 400 $$
Cantidad de producto que demanda C
$$ A_C+B_C \geq 200$$
Cantidad de producto que demanda D
$$ A_D+B_D \geq 300$$
Cantidad de producto que demanda E
$$ A_E+B_E \geq 400$$

5. Vectorizar en forma

In [77]:
import numpy as np
import scipy.optimize as opt
c3=np.array([50,25,60,40,10,20])
Aub3=np.array([
    [-1,-1,0,0,0,0],
    [0,0,-1,-1,0,0],
    [0,0,0,0,-1,-1],
])
bub3=np.array([
    [-200],
    [-300],
    [-400],
])
Aeq3=np.array([
    [1,0,1,0,1,0],
    [0,1,0,1,0,1],
])
beq3=np.array([
    [500],
    [400],
])


In [73]:
sol3=opt.linprog(c3,Aub3,bub3,Aeq3,beq3, integrality=1)
sol3

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: 23000.0
              x: [ 0.000e+00  2.000e+02  1.000e+02  2.000e+02  4.000e+02
                   0.000e+00]
            nit: -1
          lower:  residual: [ 0.000e+00  2.000e+02  1.000e+02  2.000e+02
                              4.000e+02  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf        inf        inf
                                    inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00]
          eqlin:  residual: [ 0.000e+00  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00]
        ineqlin:  residual: [ 0.000e+00  0.000e+00  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00]


In [79]:
sol3.x

array([  0., 200., 100., 200., 400.,   0.])

In [81]:
Aeq3@sol3.x

array([500., 400.])

In [83]:
Aub3@sol3.x

array([-200., -300., -400.])

Se mandan 0 en Ac, 200 en Bc, 100 en Ad, 200 en Bd y 400 en Ae

Mónica hace aretes y cadenitas de joyería. Es tan buena, que todo lo que hace lo vende.

Le toma 30 minutos hacer un par de aretes y una hora hacer una cadenita, y como Mónica también es estudihambre, solo dispone de 10 horas a la semana para hacer las joyas. Por otra parte, el material que compra solo le alcanza para hacer 15 unidades (el par de aretes cuenta como unidad) de joyas por semana.

La utilidad que le deja la venta de las joyas es $\$15$ en cada par de aretes y $\$20$ en cada cadenita.

¿Cuántos pares de aretes y cuántas cadenitas debería hacer Mónica para maximizar su utilidad?

Formular el problema en la forma explicada y obtener la solución gráfica (puede ser a mano).

1. Criterio de optimización

Maximizar la ganancia

2. Variables
   $$ x_A, x_C $$
   
3. Función de Costo
$$ Costo=15x_A+20x_C $$

4. Info adicional
Tiempo
$$ 30x_A+60x_C\leq 600 $$
Unidades
$$x_A+x_C\leq15 $$


5. Vectorizar en forma

In [17]:
import numpy as np
import scipy.optimize as opt
K=np.array([-15,-20])
AuB=np.array([
    [0.5,1],
    [1,1],
])
buB=np.array([
    [10],
    [15],
])

In [19]:
soL=opt.linprog(K,AuB,buB, integrality=1)
soL

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -250.0
              x: [ 1.000e+01  5.000e+00]
            nit: -1
          lower:  residual: [ 1.000e+01  5.000e+00]
                 marginals: [ 0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf]
                 marginals: [ 0.000e+00  0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 0.000e+00  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00]
 mip_node_count: 1
 mip_dual_bound: -250.0
        mip_gap: 0.0

In [23]:
soL.x

array([10.,  5.])

10 aretes y 5 cadenas

El objetivo de este problema es determinar la mejor estrategia de inversión, dados diferentes tipos de bono, la máxima cantidad que puede ser invertida en cada bono, el porcentaje de retorno y los años de madurez. También hay una cantidad fija de dinero disponible ($\$750,000$). Por lo menos la mitad de este dinero debe ser invertido en bonos con 10 años o más para la madurez. Se puede invertir un máximo del $25\%$ de esta cantidad en cada bono. Finalmente, hay otra restricción que no permite usar más de $35\%$ en bonos de alto riesgo.

Existen seis (6) opciones de inversión con las letras correspondientes $A_i$

1. $A_1$:(Tasa de retorno=$8.65\%$; Años para la madurez=11, Riesgo=Bajo)
1. $A_2$:(Tasa de retorno=$9.50\%$; Años para la madurez=10, Riesgo=Alto)
1. $A_3$:(Tasa de retorno=$10.00\%$; Años para la madurez=6, Riesgo=Alto)
1. $A_4$:(Tasa de retorno=$8.75\%$; Años para la madurez=10, Riesgo=Bajo)
1. $A_5$:(Tasa de retorno=$9.25\%$; Años para la madurez=7, Riesgo=Alto)
1. $A_6$:(Tasa de retorno=$9.00\%$; Años para la madurez=13, Riesgo=Bajo)

Lo que se quiere entonces es maximizar el retorno que deja la inversión.

Este problema puede ser resuelto con programación lineal. Formalmente, puede ser descrito como:

$$\max_{A_1,A_2,...,A_6}\sum^{6}_{i=1} A_iR_i,$$

donde $A_i$ representa la cantidad invertida en la opción, y $R_i$ representa la tasa de retorno respectiva.

1.Criterio de optimización

Maximizar las ganancias puestas a inversión

2. Variables
   
$A_1$:(Tasa de retorno=$8.65\%$; Años para la madurez=11, Riesgo=Bajo)

$A_2$:(Tasa de retorno=$9.50\%$; Años para la madurez=10, Riesgo=Alto)

$A_3$:(Tasa de retorno=$10.00\%$; Años para la madurez=6, Riesgo=Alto)

$A_4$:(Tasa de retorno=$8.75\%$; Años para la madurez=10, Riesgo=Bajo)

$A_5$:(Tasa de retorno=$9.25\%$; Años para la madurez=7, Riesgo=Alto)

$A_6$:(Tasa de retorno=$9.00\%$; Años para la madurez=13, Riesgo=Bajo)

4. Función costo
$$ 0.0865A_1+0.095A_2+0.1A_3+0.0875A_4+0.0925A_5+0.09A_6 =Costo$$

4. Info adicional

$$A_1+A_2+A_3+A_4+A_5+A_6\leq 750000$$

$$A_1\leq187500$$

$$A_2\leq187500$$

$$A_3\leq187500$$

$$A_4\leq187500$$

$$A_5\leq187500$$

$$A_6\leq187500$$

$$A_1+A_2+A_4+A_6\geq375000$$

$$A_2+A_3+A_5\leq262500$$

5. Vectorizar en forma

In [70]:
import numpy as np
import scipy.optimize as opt
CO=np.array([-0.0865,-0.095,-0.1,-0.0875,-0.0925,-0.09])
AUb=np.array([
    [1,1,1,1,1,1],
    [1,0,0,0,0,0],
    [0,1,0,0,0,0],
    [0,0,1,0,0,0],
    [0,0,0,1,0,0],
    [0,0,0,0,1,0],
    [0,0,0,0,0,1],
    [-1,-1,0,-1,0,-1],
    [0,1,1,0,1,0],

])
bUb=np.array([
    [750000],
    [187500],
    [187500],
    [187500],
    [187500],
    [187500],
    [187500],
    [-375000],
    [262500],
])

In [72]:
SOLL=opt.linprog(CO,AUb,bUb)
SOLL

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -68887.5
              x: [ 1.125e+05  7.500e+04  1.875e+05  1.875e+05  0.000e+00
                   1.875e+05]
            nit: 3
          lower:  residual: [ 1.125e+05  7.500e+04  1.875e+05  1.875e+05
                              0.000e+00  1.875e+05]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              2.500e-03  0.000e+00]
          upper:  residual: [       inf        inf        inf        inf
                                    inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 0.000e+00  7.500e+04  1.125e+05  0.000e+00
                              0.000e+00  1.875e+05  0.000e+00  1.875e+05
                      

In [74]:
SOLL.x

array([112500.,  75000., 187500., 187500.,      0., 187500.])

Se invierten 112500 en A1, 75000 en A2, 187500 en A3, 187500 en A4, 0 en A5 y 187500 en A6

# A05

Ejercicio 1

Se quiere producir comida para gatos de la manera más barata, no obstante se debe también asegurar que se cumplan los datos requeridos de analisis nutricional. Por lo que se quiere variar la cantidad de cada ingrediente para cumplir con los estandares nutricionales. Los requisitos que se tienen es que en 100 gramos, se deben tener por lo menos 9 gramos de proteína y  7 gramos de grasa. Así mismo, no se debe tener más de 2.5 gramos de fibra y 0.5 gramos de sal.0.0%	00.0%

![](img1.png)

1. Identificar criterio max/min

Se debe minimizar el costo maximizando la cantidad de proteínas, lípidos, fibra y sal


2. Variables de control

Se puede modificar la cantidad de alimento ya sea pollo, carne, cordero, arroz, trigo y gel
$$ x_P,x_{CA},x_{CO},x_A,x_T,x_G $$

3. Función de Costo

$$ Costo=0.00012x_P+0.00009x_{CA},+0.0001x_{CO}+0.00003x_A+0.00005x_T+0.00002x_G$$

4. Info adicional



Proteína
$$ 0.1x_P+0.2x_{CA}+0.15x_{CO}+0x_A+0.04x_T+0x_G\geq 9$$
Grasa
$$ 0.08x_P+0.1x_{CA}+0.11x_{CO}+0.01x_A+0.01x_T+0x_G\geq 7$$
Fibra
$$ 0.001x_P+0.005x_{CA}+0.005x_{CO}+0.1x_A+0.15x_T+0x_G\geq 2.5$$
Sal
$$ 0.002x_P+0.005x_{CA}+0.007x_{CO}+0.002x_A+0.008x_T+0x_G\geq 0.5$$
Peso
$$ x_P+x_{CA}+x_{CO}+x_A+x_T+x_G=100$$

5. Vectorizar en forma

In [106]:
import numpy as np
import scipy.optimize as opt
c5=np.array([0.00012,0.00009,0.0001,0.00003,0.00005,0.00002])
Aub5=np.array([
    [-0.1,-0.2,-0.15,0,-0.04,0],
    [-0.08,-0.1,-0.11,-0.01,-0.01,+0],
    [-0.001,-0.005,-0.005,-0.1,-0.15,+0],
    [-0.002,-0.005,-0.007,-0.002,-0.008,+0],
    [1,1,1,1,1,1],
])
bub5=np.array([
    [-9],
    [-7],
    [-2.5],
    [-0.5],
    [100]


In [108]:
sol5=opt.linprog(c5,Aub5,bub5, integrality=1)
sol5

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: 0.0072000000000000015
              x: [ 0.000e+00  0.000e+00  6.000e+01  4.000e+01  0.000e+00
                  -0.000e+00]
            nit: -1
          lower:  residual: [ 0.000e+00  0.000e+00  6.000e+01  4.000e+01
                              0.000e+00 -0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf        inf        inf
                                    inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00]
          eqlin:  residual: [-1.421e-14]
                 marginals: [ 0.000e+00]
        ineqlin:  residual: [ 1.776e-15  1.776e-15  1.800e+00  1.110e-16]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+0

In [116]:
Aub5@sol5.x

array([-9. , -7. , -4.3, -0.5])

In [118]:
sol5.x

array([ 0.,  0., 60., 40.,  0., -0.])

Se le pusieron 60g de cordero y 40g de arroz

In [136]:
sol5.fun

0.0072000000000000015

Ejercicio 2

La Comisión Federal de Electricidad (CFE) dispone de cuatro plantas de generación para satisfacer la demanda diaria eléctrica en cuatro ciudades, Guadalajara, León y Morelia. Las plantas 1, 2 y 3 pueden satisfacer 75, 50 y 60 millones de kW al día respectivamente. Las necesidades de las ciudades de Guadalajara, León y Morelia son de 60, 45 y 60 millones de Kw al día respectivamente.

Los costos asociados al envío de suministro energético por cada millón de kW entre cada planta y cada ciudad son los registrados en la siguiente tabla.

![](img2.png)

1. Identificar criterio max/min 
$$ \geq $$
y
$$\leq $$
2. Variables de control
$$ P_1M, P_1L, P_1G, P_2M, P_2L, P_2G ,P_3M, P_3L, P_1G   $$
3. Función de costo
$$ 7P_1M + 2P_1L + 5P_1G + 5P_2M + 4P_2L+3P_2G +3P_3M+ 2P_3L+ 5P_3G=Costo$$
4. Info adicional

Se necesita:
$$ P_1M+ P_2M+P_3M\geq 60$$
$$ P_1G+ P_2G+P_3G\geq 60$$
$$ P_1L+ P_2L+P_3L\geq 45$$
Pueden enviar
$$ P_1M+ P_1G+P_1L\leq 75$$
$$ P_2M+ P_2G+P_2L\leq 50$$
$$ P_3M+ P_3G+P_3L\leq 60$$

5. Vectorizar en forma 

In [22]:
import numpy as np
import scipy.optimize as opt
c6=np.array([7,2,5,5,4,3,3,2,5])
Aub6=np.array([
    [-1,0,0,-1,0,0,-1,0,0],
    [0,0,-1,0,0,-1,0,0,-1],
    [0,-1,0,0,-1,0,0,-1,0],
    [1,1,1,0,0,0,0,0,0],
    [0,0,0,1,1,1,0,0,0],
    [0,0,0,0,0,0,1,1,1],
])
bub6=np.array([
    [60],
    [60],
    [45],
    [75],
    [50],
    [60],
])

In [26]:
sol6=opt.linprog(c6,Aub6,bub6, integrality=1)
sol6

       message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
       success: True
        status: 0
           fun: 0.0
             x: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00  0.000e+00
                  0.000e+00  0.000e+00  0.000e+00  0.000e+00]
           nit: -1
         lower:  residual: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                             0.000e+00  0.000e+00  0.000e+00  0.000e+00
                             0.000e+00]
                marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                             0.000e+00  0.000e+00  0.000e+00  0.000e+00
                             0.000e+00]
         upper:  residual: [       inf        inf        inf        inf
                                   inf        inf        inf        inf
                                   inf]
                marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                             0.000e+00  0.000e+00  0.000e+00  0.000e+00
            

In [18]:
sol6.x

array([ 0., 45., 10.,  0.,  0., 50., 60.,  0., -0.])

La planta 1 manda 45 a Leon, 10 a Guadalajara. La planta 2 manda 50 a Leon y la planta 3 manda 60 a Morelia

In [20]:
Aub6@sol6.x

array([-60., -60., -45.,  55.,  50.,  60.])

# A06

Ejercicio 1.

Una carpintería local fabrica sillas, mesas, libreros y bases de cama. Hay dos empleados en esta carpintería: un empleado experimentado y un aprendiz. El aprendiz tarda aproximadamente 50% más tiempo que el carpintero experimentado en construir cualquier cosa, además de que tiene un desperdicio de alrededor de 20% de material. Los materiales y tiempos necesarios para construir algo se desglozan en la siguiente tabla:

![](C1.png)

Las mesas se venden en 9000 pesos, las sillas en 3000 pesos, los libreros en 10000 pesos, y las bases de cama en 9000 pesos. Suponiendo una jornada de 40 horas a la semana y que tienen 100 unidades de madera.
l?
 

2. La carpintería recibió una orden por 3 combos de mesa + 6 sillas.
   

¿Se puede cumplir la orden?
Si sí, ¿qué es lo que tienen que fabricar en el resto del tiempo y con el resto del material para maximizar su ganancia?
¿Cuántas horas trabaja cada quién?
¿Cuánto material se usa?
¿Cuál es el nuevo ingreso total?
 

1. Identificar criterio 
Se deben de maximizar las ganancias
$$\geq$$
En el menor tiempo posible
$$ \leq $$
Con el menor material
$$ \leq $$
2.Variables de control
$$ M_E,M_A,S_E,S_A,L_E,L_A,B_E,B_A$$
3. Función de costo
$$ Costo= 9000M_E+9000M_A+3000S_E+3000S_A+10000L_E+10000L_A+9000B_E+9000B_A $$
Paso 4: Info adicional

Tiempo experimentado
$$ 3M_E+1S_E+4L_E+3B_E\leq40$$

Tiempo aprendiz
$$4.5M_A+1.5S_A+6L_A+4.5B_A\leq40 $$


Material

$$ 6M_E+7.2M_A+2S_E+2.4S_A+5L_E+6L_A+5B_E+6B_A\leq100 $$


Paso 5

In [50]:
import numpy as np
import scipy.optimize as opt
c7=np.array([-9000,-9000,-3000,-3000,-10000,-10000,-9000,-9000])
Aub7=np.array([
    [3,0,1,0,4,0,3,0],
    [0,4.5,0,1.5,0,6,0,4.5],
    [6,7.2,2,2.4,5,6,5,6]
])
bub7=np.array([
    [40],
    [40],
    [100],
])


In [59]:
sol7=opt.linprog(c,Aub,bub, integrality=1)
sol7

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -176000.0
              x: [ 0.000e+00  0.000e+00  1.000e+00 -9.474e-15  9.000e+00
                   2.000e+00  1.000e+00  6.000e+00]
            nit: -1
          lower:  residual: [ 0.000e+00  0.000e+00  1.000e+00 -9.474e-15
                              9.000e+00  2.000e+00  1.000e+00  6.000e+00]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00  0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf        inf        inf
                                    inf        inf        inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00  0.000e+00  0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 0.000e+00  1.000e+00  0.00

¿Qué les conviene construir durante una semana para maximizar su ganancia?

10 sillas, 11 libreros y 7 bases de cama

In [61]:
sol7.x

array([ 0.00000000e+00,  0.00000000e+00,  1.00000000e+00, -9.47390314e-15,
        9.00000000e+00,  2.00000000e+00,  1.00000000e+00,  6.00000000e+00])

In [63]:
Aub7@sol7.x

array([ 40.,  39., 100.])

¿Cuántas horas trabaja cada quién?

Se llevan 40 el experimentado y 39 horas el aprendiz

¿Cuánto material se utilizará?

Se llevan 100 unidades de material

¿Cuál es el ingreso total si se vende todo lo fabricado?

In [75]:
sol7.fun

-176000.0

Se gana en total 176,000 pesos

2. La carpintería recibió una orden por combos de mesa + 6 sillas
Paso 1
 Se deben de maximizar las ganancias
$$\geq$$
En el menor tiempo posible
$$ \leq $$
Con el menor material
$$ \leq $$

Paso 2

$$ M_E,M_A,S_E,S_A,L_E,L_A,B_E,B_A$$

Paso 3
$$ Costo= 9000M_E+9000M_A+3000S_E+3000S_A+10000L_E+10000L_A+9000B_E+9000B_A $$

Paso 4: Info adicional

Tiempo experimentado
$$ 3M_E+1S_E+4L_E+3B_E\leq40$$

Tiempo aprendiz
$$4.5M_A+1.5S_A+6L_A+4.5B_A\leq40 $$


Material

$$ 6M_E+7.2M_A+2S_E+2.4S_A+5L_E+6L_A+5B_E+6B_A\leq100 $$

Mesa + 6 sillas
$$ M_E+M_A=3 $$
$$ S_E+S_E=18 $$

Paso 5

In [77]:
c8=np.array([-9000,-9000,-3000,-3000,-10000,-10000,-9000,-9000])
Aub8=np.array([
    [3,0,1,0,4,0,3,0],
    [0,4.5,0,1.5,0,6,0,4.5],
    [6,7.2,2,2.4,5,6,5,6],
    [-1,-1,0,0,0,0,0,0],
    [0,0,-1,-1,0,0,0,0],
    
])
bub8=np.array([
    [40],
    [40],
    [100],
    [-3],
    [-18],
])


In [79]:
sol8=opt.linprog(c8,Aub8,bub8, integrality=1)
sol8

        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -161000.0
              x: [ 3.000e+00  0.000e+00  1.800e+01  0.000e+00  3.000e+00
                   5.000e+00  0.000e+00  0.000e+00]
            nit: -1
          lower:  residual: [ 3.000e+00  0.000e+00  1.800e+01  0.000e+00
                              3.000e+00  5.000e+00  0.000e+00  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00  0.000e+00  0.000e+00]
          upper:  residual: [       inf        inf        inf        inf
                                    inf        inf        inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00  0.000e+00  0.000e+00]
          eqlin:  residual: []
                 marginals: []
        ineqlin:  residual: [ 1.000e+00  1.000e+01  1.00

In [81]:
sol8.x

array([ 3.,  0., 18.,  0.,  3.,  5.,  0.,  0.])

¿Se puede cumplir la orden?
Si sí, ¿qué es lo que tienen que fabricar en el resto del tiempo con el resto del material para maximizar su ganancia?

El experimentado 3 libreros

El aprendiz hizo 5 libreros 


In [83]:
Aub8@sol8.x

array([ 39.,  30.,  99.,  -3., -18.])

¿Cuántas horas trabaja cada quién?

Trabaja 39 el experimentado y 30 el aprendiz

¿Cuánto material se usa?
99

¿Cuál es el nuevo ingreso total?

Se gana 161,000